cmake/onnxruntime_providers.cmake

```diff
@@ -488,6 +488,39 @@ if (onnxruntime_USE_CUDA)
   endif()
 
   add_dependencies(onnxruntime_providers_cuda onnxruntime_providers_shared ${onnxruntime_EXTERNAL_DEPENDENCIES})
+
+  if (onnxruntime_ENABLE_FT)
+    file(GLOB_RECURSE onnxruntime_custom_op_ft_wrapper_srcs
+       "${ONNXRUNTIME_ROOT}/custom_ops/cuda/*.cc"
+       "${ONNXRUNTIME_ROOT}/custom_ops/cuda/*.h"
+    )
+    onnxruntime_add_shared_library_module(custom_op_ft_wrapper_library ${onnxruntime_custom_op_ft_wrapper_srcs})
+
+    target_include_directories(custom_op_ft_wrapper_library PRIVATE ${ONNXRUNTIME_ROOT}/include/onnxruntime/core/session 
+                                                                    ${onnxruntime_FT_HOME}
+                                                                    ${onnxruntime_CUDNN_HOME}/include
+                                                                    ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
+
+    set(FT_VIT_LIBS 
+      ${onnxruntime_FT_HOME}/build/lib/libViTINT8.so
```

Comment:
this doesn't need to be .so
just leave it static so you don't need to modify their cmake file.
you're already linking against other static libs here right?

Reply:
If ORT is linking to static libViTINT8.a, it will get many "symbol not found" errors when running.
It needs to link other dependencies as well (as I mentioned in the PR description).
So, I manually link them but don't know how many of them need to be linked...
      #${onnxruntime_FT_HOME}/build/lib/libtensor.a
      #${onnxruntime_FT_HOME}/build/lib/libadd_residual_kernels.a
      #${onnxruntime_FT_HOME}/build/lib/libbert_preprocess_kernels.a
      #${onnxruntime_FT_HOME}/build/lib/liblayernorm_int8_kernels.a
      #${onnxruntime_FT_HOME}/build/lib/liblayernorm_kernels.a
      #${onnxruntime_FT_HOME}/build/lib/liblayout_transformer_int8_kernels.a
      #${onnxruntime_FT_HOME}/build/lib/libUnfusedAttentionLayerINT8.a
      #${onnxruntime_FT_HOME}/build/lib/libUnfusedAttentionLayer.a
      #${onnxruntime_FT_HOME}/build/lib/libFusedAttentionLayerINT8.a
      #${onnxruntime_FT_HOME}/build/lib/libFusedAttentionLayer.a
      ...

Have not yet spent time figuring out why.

Reply:
just look at the dependencies in their cmake file:
https://github.com/NVIDIA/FasterTransformer/blob/72d3dceb97d26c1dd20d9ad9b6ab44cda2f2919a/src/fastertransformer/models/vit_int8/CMakeLists.txt#L20

Reply:
> just look at the dependencies in their cmake file: https://github.com/NVIDIA/FasterTransformer/blob/72d3dceb97d26c1dd20d9ad9b6ab44cda2f2919a/src/fastertransformer/models/vit_int8/CMakeLists.txt#L20

Thanks for pointing out, yes, it seems to work and will try it.